diff -Naur enigma2-normal/configure.ac enigma2-old/configure.ac
--- enigma2-normal/configure.ac	2021-01-05 11:51:39.211549000 +0330
+++ enigma2-old/configure.ac	2021-01-07 10:59:11.147392811 +0330
@@ -752,16 +752,15 @@
 CPPFLAGS="$CPPFLAGS $PYTHON_CPPFLAGS"
 CFLAGS="$CFLAGS -Wall"
 CXXFLAGS="$CXXFLAGS -Wall -Wextra \
-	-Wformat=2 -Wformat-overflow=2 -Wformat-nonliteral -Wformat-security \
-	-Wimplicit-fallthrough=5 -Wduplicated-branches -Wtrampolines \
-	-Wdangling-else -Wshift-overflow=2 -Wswitch \
-	-Wunused-but-set-parameter -Wunused-const-variable=1 -Wuninitialized \
-	-Wstringop-overflow=3 -Woverlength-strings \
+	-Wformat=2 -Wformat-nonliteral -Wformat-security \
+	-Wtrampolines \
+	-Wswitch \
+	-Wunused-but-set-parameter -Wuninitialized \
+	-Woverlength-strings \
 	-Wunsafe-loop-optimizations -Wpointer-arith \
 	-Wfloat-equal -Wlogical-op \
-	-Wno-error=cast-align=strict -Wno-error=ignored-qualifiers \
-	-Wno-error=cast-function-type \
-	-Wno-error=stringop-truncation -Wno-error=shadow -Wno-error=cast-qual \
+	-Wno-error=ignored-qualifiers \
+	-Wno-error=shadow -Wno-error=cast-qual \
 	-Wno-error=aggregate-return -Wno-error=missing-field-initializers \
 	-Wno-error=packed -Wno-error=vla -Wno-error=clobbered -Wno-error=unused-parameter \
 	$BASE_CFLAGS $ENIGMA2_CFLAGS $PTHREAD_CFLAGS $OPENMP_CFLAGS $ALSA_CFLAGS \
diff -Naur enigma2-normal/lib/python/python_base.i enigma2-old/lib/python/python_base.i
--- enigma2-normal/lib/python/python_base.i	2021-01-05 09:16:14.365165000 +0330
+++ enigma2-old/lib/python/python_base.i	2021-01-07 10:58:18.850127709 +0330
@@ -140,7 +140,7 @@
 	{"get", (PyCFunction)eTimerPy_get_callback_list, METH_NOARGS,
 	 "get timeout callback list"
 	},
-	{}  /* Sentinel */
+	{NULL}  /* Sentinel */
 };
 
 static PyObject *
@@ -167,22 +167,48 @@
 	 (char*)"synonym for our self",
 	 NULL},
 
-	{} /* Sentinel */
+	{NULL} /* Sentinel */
 };
 
 static PyTypeObject eTimerPyType = {
 	PyVarObject_HEAD_INIT(NULL, 0)
 	"eBaseImpl.eTimer", /*tp_name*/
 	sizeof(eTimerPy), /*tp_basicsize*/
-	.tp_dealloc = (destructor)eTimerPy_dealloc,
-	.tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
-	.tp_doc = "eTimer objects",
-	.tp_traverse = (traverseproc)eTimerPy_traverse,
-	.tp_clear = (inquiry)eTimerPy_clear,
-	.tp_weaklistoffset = offsetof(eTimerPy, in_weakreflist),
-	.tp_methods = eTimerPy_methods,
-	.tp_getset = eTimerPy_getseters,
-	.tp_new = eTimerPy_new,
+	0, /*tp_itemsize*/
+	(destructor)eTimerPy_dealloc, /*tp_dealloc*/
+	0, /*tp_print*/
+	0, /*tp_getattr*/
+	0, /*tp_setattr*/
+	0, /*tp_compare*/
+	0, /*tp_repr*/
+	0, /*tp_as_number*/
+	0, /*tp_as_sequence*/
+	0, /*tp_as_mapping*/
+	0, /*tp_hash */
+	0, /*tp_call*/
+	0, /*tp_str*/
+	0, /*tp_getattro*/
+	0, /*tp_setattro*/
+	0, /*tp_as_buffer*/
+	Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC, /*tp_flags*/
+	"eTimer objects", /* tp_doc */
+	(traverseproc)eTimerPy_traverse, /* tp_traverse */
+	(inquiry)eTimerPy_clear, /* tp_clear */
+	0, /* tp_richcompare */
+	offsetof(eTimerPy, in_weakreflist), /* tp_weaklistoffset */
+	0, /* tp_iter */
+	0, /* tp_iternext */
+	eTimerPy_methods, /* tp_methods */
+	0, /* tp_members */
+	eTimerPy_getseters, /* tp_getset */
+	0, /* tp_base */
+	0, /* tp_dict */
+	0, /* tp_descr_get */
+	0, /* tp_descr_set */
+	0, /* tp_dictoffset */
+	0, /* tp_init */
+	0, /* tp_alloc */
+	eTimerPy_new, /* tp_new */
 };
 
 // eSocketNotifier replacement
@@ -311,7 +337,7 @@
 	{"setRequested", (PyCFunction)eSocketNotifierPy_set_requested, METH_VARARGS,
 	 "set requested"
 	},
-	{}  /* Sentinel */
+	{NULL}  /* Sentinel */
 };
 
 static PyObject *
@@ -325,26 +351,52 @@
 	 (getter)eSocketNotifierPy_get_cb_list, (setter)0,
 	 (char*)"returns the callback python list",
 	 NULL},
-	{} /* Sentinel */
+	{NULL} /* Sentinel */
 };
 
 static PyTypeObject eSocketNotifierPyType = {
 	PyVarObject_HEAD_INIT(NULL, 0)
 	"eBaseImpl.eSocketNotifier", /*tp_name*/
 	sizeof(eSocketNotifierPy), /*tp_basicsize*/
-	.tp_dealloc = (destructor)eSocketNotifierPy_dealloc,
-	.tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
-	.tp_doc = "eSocketNotifier objects",
-	.tp_traverse = (traverseproc)eSocketNotifierPy_traverse,
-	.tp_clear = (inquiry)eSocketNotifierPy_clear,
-	.tp_weaklistoffset = offsetof(eSocketNotifierPy, in_weakreflist),
-	.tp_methods = eSocketNotifierPy_methods,
-	.tp_getset = eSocketNotifierPy_getseters,
-	.tp_new = eSocketNotifierPy_new,
+	0, /*tp_itemsize*/
+	(destructor)eSocketNotifierPy_dealloc, /*tp_dealloc*/
+	0, /*tp_print*/
+	0, /*tp_getattr*/
+	0, /*tp_setattr*/
+	0, /*tp_compare*/
+	0, /*tp_repr*/
+	0, /*tp_as_number*/
+	0, /*tp_as_sequence*/
+	0, /*tp_as_mapping*/
+	0, /*tp_hash */
+	0, /*tp_call*/
+	0, /*tp_str*/
+	0, /*tp_getattro*/
+	0, /*tp_setattro*/
+	0, /*tp_as_buffer*/
+	Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC, /*tp_flags*/
+	"eTimer objects", /* tp_doc */
+	(traverseproc)eSocketNotifierPy_traverse, /* tp_traverse */
+	(inquiry)eSocketNotifierPy_clear, /* tp_clear */
+	0, /* tp_richcompare */
+	offsetof(eSocketNotifierPy, in_weakreflist), /* tp_weaklistoffset */
+	0, /* tp_iter */
+	0, /* tp_iternext */
+	eSocketNotifierPy_methods, /* tp_methods */
+	0, /* tp_members */
+	eSocketNotifierPy_getseters, /* tp_getset */
+	0, /* tp_base */
+	0, /* tp_dict */
+	0, /* tp_descr_get */
+	0, /* tp_descr_set */
+	0, /* tp_dictoffset */
+	0, /* tp_init */
+	0, /* tp_alloc */
+	eSocketNotifierPy_new, /* tp_new */
 };
 
 static PyMethodDef base_module_methods[] = {
-	{}  /* Sentinel */
+	{NULL}  /* Sentinel */
 };
 
 #if PY_MAJOR_VERSION >= 3
diff -Naur enigma2-normal/lib/python/python_console.i enigma2-old/lib/python/python_console.i
--- enigma2-normal/lib/python/python_console.i	2021-01-05 09:16:14.365165000 +0330
+++ enigma2-old/lib/python/python_console.i	2021-01-07 10:58:18.850127709 +0330
@@ -62,7 +62,7 @@
 	 (getter)eConsolePy_appClosed, (setter)0,
 	 (char*)"appClosed callback list",
 	 NULL},
-	{} /* Sentinel */
+	{NULL} /* Sentinel */
 };
 
 static int
@@ -331,26 +331,52 @@
 	{(char*)"running", (PyCFunction)eConsolePy_running, METH_NOARGS,
 	 (char*)"returns the running state"
 	},
-	{}  /* Sentinel */
+	{NULL}  /* Sentinel */
 };
 
 static PyTypeObject eConsolePyType = {
 	PyVarObject_HEAD_INIT(NULL, 0)
 	"eConsoleImpl.eConsoleAppContainer", /*tp_name*/
 	sizeof(eConsolePy), /*tp_basicsize*/
-	.tp_dealloc = (destructor)eConsolePy_dealloc,
-	.tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
-	.tp_doc = "eConsoleAppContainer objects",
-	.tp_traverse = (traverseproc)eConsolePy_traverse,
-	.tp_clear = (inquiry)eConsolePy_clear,
-	.tp_weaklistoffset = offsetof(eConsolePy, in_weakreflist),
-	.tp_methods = eConsolePy_methods,
-	.tp_getset = eConsolePy_getseters,
-	.tp_new = eConsolePy_new,
+	0, /*tp_itemsize*/
+	(destructor)eConsolePy_dealloc, /*tp_dealloc*/
+	0, /*tp_print*/
+	0, /*tp_getattr*/
+	0, /*tp_setattr*/
+	0, /*tp_compare*/
+	0, /*tp_repr*/
+	0, /*tp_as_number*/
+	0, /*tp_as_sequence*/
+	0, /*tp_as_mapping*/
+	0, /*tp_hash */
+	0, /*tp_call*/
+	0, /*tp_str*/
+	0, /*tp_getattro*/
+	0, /*tp_setattro*/
+	0, /*tp_as_buffer*/
+	Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC, /*tp_flags*/
+	"eConsoleAppContainer objects", /* tp_doc */
+	(traverseproc)eConsolePy_traverse, /* tp_traverse */
+	(inquiry)eConsolePy_clear, /* tp_clear */
+	0, /* tp_richcompare */
+	offsetof(eConsolePy, in_weakreflist), /* tp_weaklistoffset */
+	0, /* tp_iter */
+	0, /* tp_iternext */
+	eConsolePy_methods, /* tp_methods */
+	0, /* tp_members */
+	eConsolePy_getseters, /* tp_getset */
+	0, /* tp_base */
+	0, /* tp_dict */
+	0, /* tp_descr_get */
+	0, /* tp_descr_set */
+	0, /* tp_dictoffset */
+	0, /* tp_init */
+	0, /* tp_alloc */
+	eConsolePy_new, /* tp_new */
 };
 
 static PyMethodDef console_module_methods[] = {
-	{}  /* Sentinel */
+	{NULL}  /* Sentinel */
 };
 
 #if PY_MAJOR_VERSION >= 3
