diff -Naur meson-0.56.0/mesonbuild/environment.py meson-0.56.0-PersianPrince/mesonbuild/environment.py
--- meson-0.56.0/mesonbuild/environment.py	2020-10-30 11:47:30.000000000 +0330
+++ meson-0.56.0-PersianPrince/mesonbuild/environment.py	2020-12-17 20:11:26.958636104 +0330
@@ -374,8 +374,11 @@
         trial = 'x86_64'
     elif trial in {'sun4u', 'sun4v'}:
         trial = 'sparc64'
-    elif trial in {'mipsel', 'mips64el'}:
-        trial = trial.rstrip('el')
+    elif trial.startswith('mips'):
+        if not '64' in trial:
+            trial = 'mips'
+        else:
+            trial = 'mips64'
     elif trial in {'ip30', 'ip35'}:
         trial = 'mips64'
 
@@ -430,7 +433,10 @@
         # Make more precise CPU detection for Elbrus platform.
         trial = platform.processor().lower()
     elif trial.startswith('mips'):
-        trial = trial.rstrip('el')
+        if not '64' in trial:
+            trial = 'mips'
+        else:
+            trial = 'mips64'
 
     # Add more quirks here as bugs are reported. Keep in sync with
     # detect_cpu_family() above.
