diff -Naur gstreamer-1.18.2/gst/gstsystemclock.c gstreamer-1.18.2-PersianPrince/gst/gstsystemclock.c
--- gstreamer-1.18.2/gst/gstsystemclock.c	2020-12-06 16:51:21.621347000 +0330
+++ gstreamer-1.18.2-PersianPrince/gst/gstsystemclock.c	2020-12-18 12:34:45.059500364 +0330
@@ -70,7 +70,7 @@
 #define GST_SYSTEM_CLOCK_WAIT(clock)            g_cond_wait(GST_SYSTEM_CLOCK_GET_COND(clock),GST_SYSTEM_CLOCK_GET_LOCK(clock))
 #define GST_SYSTEM_CLOCK_BROADCAST(clock)       g_cond_broadcast(GST_SYSTEM_CLOCK_GET_COND(clock))
 
-#if defined(HAVE_FUTEX)
+#if !defined(HAVE_FUTEX)
 #include <unistd.h>
 #include <linux/futex.h>
 #include <sys/syscall.h>
