--- git/blindscan-s2.h.orig	2017-08-24 15:45:20.871303083 +0200
+++ git/blindscan-s2.h	2017-08-24 15:45:48.077638279 +0200
@@ -28,6 +28,14 @@
 #include <linux/dvb/frontend.h>
 #include "diseqc.h"
 
+#ifndef NO_STREAM_ID_FILTER
+#define NO_STREAM_ID_FILTER    (~0U)
+#endif
+
+#ifndef DTV_STREAM_ID
+#define DTV_STREAM_ID DTV_ISDBS_TS_ID
+#endif
+
 #define SLEEP_TIME 20000
 #define HORIZONTAL SEC_VOLTAGE_18
 #define VERTICAL SEC_VOLTAGE_13
